generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -172,6 +173,102 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
         depth, fuzzer)));
   }
 
+  /**
+   * Function to generate an opaque zero or one from the determinant of a generated matrix.
+   * This function heavily relies on the fact that a triangular matrix - a matrix that only has
+   * values below the diagonal (lower triangular) or above the diagonal (upper triangular) - always
+   * has a determinant that is the product of its diagonal values. For example,
```

Comment:
I did a little digging and it seems that the trace is the *sum* of the diagonal, rather than the product. Apparently it's related to the derivative of the determinant somehow, but not the determinant itself.

I definitely should have been more clear on 'its diagonals', though - thanks.

Reply:
Good catch on 'trace' - thanks.